src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -75,7 +76,76 @@ internal static int CompareStringIgnoreCaseNonAscii(ref char strA, int lengthA,
             return OrdinalCasing.CompareStringIgnoreCase(ref strA, lengthA, ref strB, lengthB);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool Vector128EqualsIgnoreCaseAscii(Vector128<sbyte> vec1, Vector128<sbyte> vec2)
+        {
+            Debug.Assert(Vector128AllAscii(vec1.AsUInt16()));
+            Debug.Assert(Vector128AllAscii(vec2.AsUInt16()));
```

Comment:
Are these needed?
This could isn't executed if the check in L113 fails.

Reply:
It is needed in case if someone else decides to use it. Scalar paths also have these asserts